src/test/validatorTest.ts

```diff
@@ -1400,4 +1408,333 @@ describe('validator', () => {
             
         })
     });
+
+    describe('type inference unit tests', () => {
+
+        describe('inferPrimitiveValueType', () => {
+
+          it('should be able to infer an Integer value - typeof number', () => {
+              let input = 101;
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Integer');
+          });
+
+          it('should be able to infer an Integer value - typeof string', () => {
+              let input = '101';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Integer');
+          });
+
+          it('should be able to infer a Double value - typeof number', () => {
+              let input = 101.01;
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Double');
+          });
+
+          it('should be able to infer a Double value - typeof string', () => {
+              let input = '101.01';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Double');
+          });
+
+          it('should be able to infer a Boolean value - typeof boolean', () => {
+              let input = false;
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Boolean');
+          });
+
+          it('should be able to infer a Boolean value - typeof string', () => {
+              let input = 'false';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Boolean');
+          });
+
+          it('should be able to infer a Json value - typeof object', () => {
+              let input = {};
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Json');
+          });
+
+          it('should be able to infer a Json value - typeof string', () => {
+              let input = '{}';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('Json');
+          });
+
+          it('should be able to infer a String value', () => {
+              let input = 'somethingBeautiful';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
+              expect(result).to.equal('String');
+          });
+
+          it('should be able to infer a Timestamp value', () => {
+              let input = '2012-12-30T20:12:22.222222-08:00';
+              let result = validator.__TESTING__.inferPrimitiveValueType(input);
```

Comment:
Might be worth testing with more timestamp ISO8601 types here if we see it being used?

For ex. no microseconds, and a zulu timezone etc

Reply:
I'm not really sure because I've been mainly targeting the integration that occurs within the `inferPrimitiveValueType` function and relied upon the functional tests for each specific function, such as `isTimestamp`, to be exhaustive. 
Please let me know if you think it should be done differently.